### PR TITLE
Added default arguments to Visibility functions

### PR DESCRIPTION
**Pull request by metanoid**
Lack of default arguments was confusing lintr

### CODE REVIEW

**Comment by daattali** [comment]
I've never used lintr..  @jimhester do I really have to have to have default arguments for every function?  I don't really want to be forced to have default `NULL` for every single argument, seems verbose and unnecessary

**Comment by jimhester** [comment]
No you should not need to have default arguments.  Not sure what is going on, I will try to look into it.

**Comment by daattali** [comment]
thanks @jimhester, whenever you have time (obviously this week is a bit busy). @metanoid can you provide more info on what happened with lintr and what gets fixed by having default args?

**Comment by daattali** [comment]
Never really noticed those warnings so I'll look into that.  It might have
to wait a few days since I'm at a conference right now, but I'll take a look

**Comment by jimhester** [comment]
Ah I see `hide()` uses `match.call()` to capture the call and pass it to a helper function.

The warnings are coming from `codetools::checkUsage()` which parses the AST to find unused arguments. It isn't going to be able to find where those arguments are used due to the meta-programming, so you will either have to live with the warnings or set default values.

**Comment by daattali** [comment]
Personally I'd rather live with the warnings, unless you (@jimhester) have an opinion about this?

**Comment by jimhester** [comment]
Another option would be to to add [lint exclusions](https://github.com/jimhester/lintr/blob/master/R/exclude.R#L9-L12) to the lines showing the warnings to silence them.

In that case no changes would need to be made here,  @metanoid would simply need to add `# nolint` comments to the appropriate lines.

**Comment by daattali** [comment]
thanks @jimhester

**Comment by metanoid** [comment]
I can add no lint comments, I can add NULL arguments to each parameter in the function call, or I can ignore the warnings.  These are all workable solutions for my specific case, thanks.
I agree that adding NULL defaults to all arguments would be an unnecessary hassle.

**Comment by daattali** [comment]
Adding the no lint comments is my preferred solution.  I'll accept that PR
if you make one:)

**Comment by metanoid** [comment]
@daattali I don't think it works that way.  I've tested adding nolint comments to the shinyjs code
([here](https://github.com/metanoid/shinyjs/tree/no-lint)) and while it still builds fine, lintr (via RStudio's Show Diagnostics) picks up the syntax warnings in my shiny app's code.  I think the only way to use nolint comments is to insert them wherever the user has actually used the shinyjs functions in question.  Unless I'm misunderstanding.

**Comment by jimhester** [comment]
Right the exclusions would need to be added to the calls @metanoid's is using, as that is where the warnings are being generated.

**Comment by metanoid** [comment]
I'm happy to do so.  You can ignore or close this PR for now.

**Comment by daattali** [comment]
Thanks for the help @jimhester 

Closing the PR

File: R/jsFunc-stateFuncs.R [hunk 1]
```diff
@@ -75,23 +75,23 @@ NULL
 
 #' @export
 #' @rdname stateFuncs
-enable <- function(id, selector) {
+enable <- function(id = NULL, selector = NULL) {
```
Comment (daattali) [inline R/jsFunc-stateFuncs.R]: test

File: R/jsFunc-stateFuncs.R [hunk 2]
```diff
@@ -75,23 +75,23 @@ NULL
...
   fxn <- "enable"
   params <- as.list(match.call())[-1]
   jsFuncHelper(fxn, params)
 }
 
 #' @export
 #' @rdname stateFuncs
```
Comment (daattali) [inline R/jsFunc-stateFuncs.R]: asdfdf

File: R/jsFunc-stateFuncs.R [hunk 3]
```diff
@@ -75,23 +75,23 @@ NULL
...
-disable <- function(id, selector) {
+disable <- function(id = NULL, selector = NULL) {
```
Comment (daattali) [inline R/jsFunc-stateFuncs.R]: lk

File: R/jsFunc-stateFuncs.R (hunk 1 above)
Comment (daattali) [inline R/jsFunc-stateFuncs.R]: kjhkj

**Pull request closed without merging**